hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/keys/CopyKeyHandler.java

```diff
@@ -89,17 +89,11 @@ protected void execute(OzoneClient client, OzoneAddress address)
               OZONE_REPLICATION_TYPE_DEFAULT));
     }
 
-    Map<String, String> keyMetadata = new HashMap<>();
-    String gdprEnabled = bucket.getMetadata().get(OzoneConsts.GDPR_FLAG);
-    if (Boolean.parseBoolean(gdprEnabled)) {
-      keyMetadata.put(OzoneConsts.GDPR_FLAG, Boolean.TRUE.toString());
-    }
-
     int chunkSize = (int) getConf().getStorageSize(OZONE_SCM_CHUNK_SIZE_KEY,
         OZONE_SCM_CHUNK_SIZE_DEFAULT, StorageUnit.BYTES);
     try (InputStream input = bucket.readKey(fromKey);
-         OutputStream output = bucket.createKey(toKey, 0,
-             replicationType, replicationFactor, keyMetadata)) {
+         OutputStream output = bucket.createKey(toKey, input.available(),
+             replicationType, replicationFactor, bucket.getMetadata())) {
```

Comment:
Your previous code was fine. The bucket metadata object may have properties that may not be needed for the key. So, if we copy the bucket metadata to be saved as key metadata, that is basically the piece of data which is mostly irrelevant for the key.

So you can replace `bucket.getMetadata()` with `keyMetadata`.